torchdynamo/optimizations/normalize.py

```diff
@@ -429,8 +429,11 @@ def normalize_ir(gm, example_inputs):
         except AttributeError:
             # log.exception("NormalizeOperators() failed")
             pass
-        ShapeAliasingAndMutationProp(gm).run(*example_inputs)
-        gm = Functionalization(gm).transform()
+        import functorch._src.config
+
+        if not functorch._src.config.use_functionalize:
```

Comment:
I think this check should go here instead:
https://github.com/pytorch/torchdynamo/blob/bb60ba951dcccb8de5aef6e930ef6cc7a30b24e3/torchdynamo/optimizations/training.py#L32

So it only affect AOT Autograd (and not other backends).